api/users/tests/utils/test_mailer_lite.py

```diff
@@ -0,0 +1,161 @@
+import json
+
+import pytest
+
+from organisations.models import Organisation, Subscription
+from users.models import FFAdminUser
+from users.utils.mailer_lite import (
+    BatchSubscribe,
+    MailerLite,
+    MailerLiteBaseClient,
+    _get_request_body_from_user,
+)
+
+
+@pytest.mark.django_db
+def test_mailer_lite_subscribe_calls_post_with_correct_arguments(mocker, settings):
+    # Given
+    mocked_requests = mocker.patch("users.utils.mailer_lite.requests")
+    base_url = "http//localhost/mailer/test/"
+    settings.MAILERLITE_BASE_URL = base_url
+    resource = "subscribers"
+
+    user = FFAdminUser.objects.create(
+        email="test_user", first_name="test", last_name="test", is_subscribed=True
+    )
+    mailer_lite = MailerLite()
+    mocked_headers = mocker.patch(
+        "users.utils.mailer_lite.MailerLiteBaseClient._get_request_headers"
+    ).return_value
+    # When
+    mailer_lite._subscribe(user)
+    # Then
+    mocked_requests.post.assert_called_with(
+        base_url + resource,
+        data=json.dumps(
+            {"email": user.email, "name": "test test", "fields": {"is_paid": False}}
+        ),
+        headers=mocked_headers,
+    )
+
+
+@pytest.mark.django_db
+def test_batch_subscribe__subscribe_calls_batch_send_correct_number_of_times(mocker):
+    # Given
+    user1 = FFAdminUser.objects.create(
+        email="test_user1", first_name="test", last_name="test"
+    )
+    user2 = FFAdminUser.objects.create(
+        email="test_user2", first_name="test", last_name="test"
+    )
+    user3 = FFAdminUser.objects.create(
+        email="test_user3", first_name="test", last_name="test"
+    )
+
+    users = [user1, user2, user3]
+    mocker.patch("users.utils.mailer_lite.MAX_BATCH_SIZE", 2)
+    mocked_batch_send = mocker.patch(
+        "users.utils.mailer_lite.BatchSubscribe._batch_send"
```

Comment:
Wouldn't it be better to just mock the `requests` module again and check that the correct number of API requests are made?